File: dotnet-adapters-shortest-paths-core/main/java/com/programmerare/shortestpaths/core/impl/EdgeImpl.java
/*
* Copyright (c) Tomas Johansson , http://www.programmerare.com
* The code is made available under the terms of the MIT License.
* https://github.com/TomasJohansson/adapters-shortest-paths/blob/master/adapters-shortest-paths-core/License.txt
*/
package com.programmerare.shortestpaths.core.impl;

import com.programmerare.shortestpaths.core.api.Edge;
import com.programmerare.shortestpaths.core.api.Vertex;
import com.programmerare.shortestpaths.core.api.Weight;
import com.programmerare.shortestpaths.core.impl.generics.EdgeGenericsImpl;

/**
 * @author Tomas Johansson
 */
public final class EdgeImpl extends EdgeGenericsImpl<Vertex, Weight> implements Edge {

	protected EdgeImpl(
		final String edgeId, 
		final Vertex startVertex, 
		final Vertex endVertex, 
		final Weight weight
	) {
		super(edgeId, startVertex, endVertex, weight);
	}

	public static Edge createEdge(
		final Vertex startVertex, 
		final Vertex endVertex, 
		final Weight weight
	) {
		final String edgeId = createEdgeIdValue(startVertex.getVertexId(), endVertex.getVertexId());
		return createEdge(edgeId, startVertex, endVertex, weight);
	}

	public static Edge createEdge(
		final String edgeId, 
		final Vertex startVertex, 
		final Vertex endVertex, 
		final Weight weight
	) {
		return new EdgeImpl(edgeId, startVertex, endVertex, weight);
	}	
}

File: dotnet-adapters-shortest-paths-core/main/com.programmerare.shortestpaths/core/impl/EdgeImpl.cs
/*
* Copyright (c) Tomas Johansson , http://www.programmerare.com
* The code is made available under the terms of the MIT License.
* https://github.com/TomasJohansson/adapters-shortest-paths/blob/master/adapters-shortest-paths-core/License.txt
*/
using com.programmerare.shortestpaths.core.api;
using com.programmerare.shortestpaths.core.impl.generics;

namespace com.programmerare.shortestpaths.core.impl
{
    /**
     * @author Tomas Johansson
     */
    public sealed class EdgeImpl : EdgeGenericsImpl<Vertex, Weight> , Edge {

	    private EdgeImpl(
		    string edgeId, 
		    Vertex startVertex, 
		    Vertex endVertex, 
		    Weight weight
	    ) : base(edgeId, startVertex, endVertex, weight) {
	    }

	    public static Edge CreateEdge(
		    Vertex startVertex, 
		    Vertex endVertex, 
		    Weight weight
	    ) {
		    string edgeId = CreateEdgeIdValue(startVertex.VertexId, endVertex.VertexId);
		    return CreateEdge(edgeId, startVertex, endVertex, weight);
	    }

	    public static Edge CreateEdge(
		    string edgeId, 
		    Vertex startVertex, 
		    Vertex endVertex, 
		    Weight weight
	    ) {
		    return new EdgeImpl(edgeId, startVertex, endVertex, weight);
	    }	
    }
}